File: Unidad 3-Funciones/Ejercicio8.c
#include <stdio.h>

//Prototipos de las funciones
void leerEntrada (int *);
int compararNumeros (int, int);
void imprimirSalida (int);


int main() {
    int numeroUno = 0, numeroDos = 0, caso = 0;

    leerEntrada(&numeroUno);
    leerEntrada(&numeroDos);
    caso = compararNumeros(numeroUno, numeroDos);
    imprimirSalida(caso);

    return 0;
}

void leerEntrada(int *_entrada){
    scanf ("%d", _entrada);
}

int compararNumeros (int a, int b){
    if(a%b == 0){
            return 1;
        } else {
            return 2;
        }
}

void imprimirSalida(int c){
    switch (c)
    {
    case 1:
        printf("Si es multiplo");
        break;
    
    case 2:
        printf("No es multiplo");
        break;   
    }
}

File: Unidad 3-Funciones/Ejercicio8.java
/*
Autor: Pedro Cauich

Problema:Escribe un programa que reciba 2 números y te imprima cual es el número
mayor, cual es el número menor o si son iguales en caso de que sea
necesario. 
 */
import java.util.Scanner;

 public class Main{
     public static void main(String[] args) {
         //Se hace el scan de la entrada, donde el valor de ella se mete al procesamiento, y ese valor resultante entra a la salida
   
        int numeroUno = leerNumero();
        int numeroDos = leerNumero();
        int caso = procesamiento(numeroUno, numeroDos);
        salida(caso); 
     }
     
     public static int leerNumero(){
        Scanner entra1 = new Scanner(System.in); //Declaramos "entra1" como objeto de entrada
        int num1 = entra1.nextInt(); //Leemos el dato, num1 es una variable local
        return num1; //Regresamos el valor de la funcion
     }
     public static int procesamiento(int x, int y){
        if(x%y == 0){
            return 1;
        }else {
            return 2;
        }
     }
     public static void salida(int c){

         switch (c){
             case 1:
             System.out.println("Si es multiplo");
             break;
             case 2:
             System.out.println("No es multiplo");
         }
     }
 }